### PR TITLE
Add App Autoscaler Feature

### PR DESCRIPTION
**Pull request by Proplex**
This PR adds the ability for an operate to co-locate an [App Autoscaler](https://github.com/cloudfoundry-incubator/app-autoscaler) as a service broker to dynamically scale apps depending on policies provided.

The feature is `autoscaler`, but also brings along with it some general kit-related changes to make it easier for operators to deploy an App Autoscaler instance. Mostly, this relates to DB selection as AA requires a PostgreSQL DB. How AA selects a DB to use is outlined in `MANUAL.md` L405.

Documentation has been written, and this has been tested in vSphere with internal (HA/non-HA) PostgreSQL, external MySQL, and external PostgreSQL options chosen. All successfully deploy.

This release also relies on a non-yet-released boshrelease and as such relies on a self-hosted S3 blob with a `dev` tag.

### CODE REVIEW

File: MANUAL.md [hunk 1]
```diff
@@ -382,6 +382,38 @@ NFS volumes provided by the NFS Volume Services Broker.
...
+
+  - `autoscaler_db_name` - Name of the SQL database name for autoscaler to use. Defaults to `autoscalerdb`. Only used if external PostgreSQL is selected for CF.
+
+  - `autoscaler_broker_url` - URL to register with the Route Register. Defaults to `autoscalerservicebroker.system.basedomain.com`
+
+  - `autoscaler_plans` - A YAML list of plans for the service broker. Defaults to:
+  ```
+    - id: autoscaler-example-plan-id
+    name: autoscaler-example-plan
+    description: This is the example service plan for the Auto-Scaling service.
```
Comment (jhunt) [inline MANUAL.md]: I think the whitespace is off here.

File: MANUAL.md [hunk 2]
```diff
@@ -382,6 +382,38 @@ NFS volumes provided by the NFS Volume Services Broker.
...
+
+If you wish to dynamically scale your instances based on pre-defined policies via Cloud Foundry's [App Autoscaler](https://github.com/cloudfoundry-incubator/app-autoscaler), you can do so via the `autoscaler` feature. It acts as a service broker, and must be bound to your organization & space. The following parameters are configurable:
+
+  - `autoscaler_instances` - How many instances to deploy of Autoscaler itself. Defaults to `1`
+  - `autoscaler_network` - Which network to deploy Autoscaler on. Defaults to `cf-autoscaler`
+  - `autoscaler_vm_type` - Which VM type to use for Autoscaler VMs. These can be low-CPU, small-disk VMs. Defaults to `default`
+
+  - `autoscaler_db_name` - Name of the SQL database name for autoscaler to use. Defaults to `autoscalerdb`. Only used if external PostgreSQL is selected for CF.
+
+  - `autoscaler_broker_url` - URL to register with the Route Register. Defaults to `autoscalerservicebroker.system.basedomain.com`
```
Comment (jhunt) [inline MANUAL.md]: Use `$system_domain` instead of `basedomain.com`.

File: hooks/post-deploy [hunk 3]
```diff
@@ -18,5 +18,11 @@ if [[ $GENESIS_DEPLOY_RC == 0 ]]; then
       echo
   describe "  #G{genesis do $GENESIS_ENVIRONMENT -- login}"
       echo
+  if [[ $(exodus autoscaler_enabled) == "1" ]]; then
+      echo "To bind the App Autoscaler service broker, run"
+      echo
+  describe "  #G{genesis info $GENESIS_ENVIRONMENT -- bind-autoscaler}"
```
Comment (jhunt) [inline hooks/post-deploy]: This should be `genesis do`

File: manifests/addons/app-autoscaler.yml [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+- name: app-autoscaler
+  url: https://app-autoscaler-boshrelease.s3.amazonaws.com/app-autoscaler.tgz
+  sha1: 7be27f150828f3f09499c72ef18dce4b4f5bd18d
+  version: 0+dev.4
+- name: consul
+  version: latest
+- name: routing
+  url: https://bosh.io/d/github.com/cloudfoundry-incubator/cf-routing-release?v=0.162.0
+  sha1: f17cf09d2414f5f486d18bbd57b17fd48fb69773
+  version: '0.162.0'
```
Comment (jhunt) [inline manifests/addons/app-autoscaler.yml]: We should probably not need to list the routing or consul releases explicitly, since those are core.

**Comment by Proplex** [comment]
All donzo

**Pull request merged**